src/client/debugger/extension/hooks/childProcessAttachService.ts

```
@@ -33,7 +33,7 @@ export class ChildProcessAttachService implements IChildProcessAttachService {
         let debugConfig: AttachRequestArguments & DebugConfiguration;
         let processId: number;
         if (data.rootStartRequest) {
-            processId = data.processId;
+            processId = data.processId!;
```

Not sure about this `!`
I'd throw an exception of it is undefined. Rather than letting it go all the way to ptvsd and then having that throw exceptions for these edge cases...

---

`data` here refers to a `ChildProcessLaunchData` instance, which has a `processId` key. However the function param is `data: ChildProcessLaunchData | (AttachRequestArguments & DebugConfiguration)`, and `processId` is optional in `AttachRequestArguments`.

Removing the force-unwrapping and casting `data` to `ChildProcessLaunchData` instead.